src/Microsoft.AspNetCore.Server.Kestrel/Http/Frame.cs

```diff
@@ -234,14 +234,13 @@ public void Reset()
             StatusCode = 200;
             ReasonPhrase = null;
 
-            var httpConnectionFeature = this as IHttpConnectionFeature;
-            httpConnectionFeature.RemoteIpAddress = RemoteEndPoint?.Address;
-            httpConnectionFeature.RemotePort = RemoteEndPoint?.Port ?? 0;
+            RemoteIpAddress = RemoteEndPoint?.Address;
+            RemotePort = RemoteEndPoint?.Port ?? 0;
 
-            httpConnectionFeature.LocalIpAddress = LocalEndPoint?.Address;
-            httpConnectionFeature.LocalPort = LocalEndPoint?.Port ?? 0;
+            LocalIpAddress = LocalEndPoint?.Address;
+            LocalPort = LocalEndPoint?.Port ?? 0;
 
-            httpConnectionFeature.ConnectionId = ConnectionId;
```

Comment:
Ok. Then I think the way to avoid the the virtual set, is to put all the `IHttpConnectionFeature` related auto getter-setter properties in Frame.cs, and put pass through interface-specific properties in Frame.FeatureCollection.cs.

We will need a new property in `Frame.cs` called something like `ConnectionIdFeature`. E.g:

        string IHttpConnectionFeature.ConnectionId
        {
            get
            {
                return ConnectionIdFeature ?? ConnectionId;
            }

            set
            {
                ConnectionIdFeature = value;
            }
        }
 
Then you can just set `ConnectionIdFeature` to null in `Reset()`.